packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -258,4 +304,225 @@ describe('Select', () => {
             expect(getSelectedValueTitle()).toBe('');
         });
     });
+
+    describe('filtering', () => {
```

Comment:
Could you make sure we have tests that verify each of these cases? I've noticed issues with some of them at various iterations of the implementation and if those issues are worth fixing then they're worth testing for. Others are things I noticed in the implementation but didn't see covered.

1. which keys can be used to open the dropdown
1. the focus state of the filter input after opening the dropdown both via the mouse and via the keyboard
1. the focus visible state of the select after various operations which close the drop down (Enter, Esc, click on an option, click outside) with both filter modes
1. the ARIA roles/states of the component and its parts
1. other attributes that are configured in the template like the placeholder text and disabled state of the filter input
1. changing the slotted options while the filter is applied should apply the filter to the new options
1. clicking in the filter input shouldn't close the dropdown
2. space should select an item if filterMode is none

Reply:
I've added tests to cover _most_ of what you've identified as gaps. The one thing that I still have not found a way to successfully test is the focus state of the filter input after opening the dropdown. For now I suggest we move forward and just concede that there doesn't seem to be a practical way to test this (or if there is it's not worth delaying the feature to discover what the proper means is).